### PR TITLE
notification schedule settings

### PR DESCRIPTION
**Pull request by DimaSannytski**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: dao/src/main/resources/db/changelog/logs/ch-add-table-notification-schedule-Sannytski.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet id="Sannytski-1" author="Dmytro Sannytski">
+        <createTable tableName="notification_schedule">
+            <column name="notification_type" type="VARCHAR" autoIncrement="false">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="seconds" type="VARCHAR(10)">
```
Comment (thesvider) [inline dao/src/main/resources/db/changelog/logs/ch-add-table-notification-schedule-Sannytski.xml]: doesn't make sense to have separate columns for each character in cron expression. keep just one column.

**Review by thesvider — CHANGES_REQUESTED** [review]
resolve comments

**Review by RossSihovsk — APPROVED** [review]

**Review by KuzbytMaksym — APPROVED** [review]

File: service/src/main/java/greencity/service/NotificationServiceImpl.java [hunk 2]
```diff
@@ -382,4 +348,17 @@ public static NotificationDto createNotificationDto(UserNotification notificatio
         return NotificationDto.builder().title(template.getTitle())
             .body(resultBody).build();
     }
+
+    private void fillAdnSendNotification(Set<NotificationParameter> parameters, Order order,
+        NotificationType notificationType) {
+        UserNotification userNotification = new UserNotification();
+        userNotification.setNotificationType(notificationType);
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/NotificationServiceImpl.java:355]: Can we use here model mapper ?

File: service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java [hunk 3]
```diff
@@ -45,7 +54,9 @@ public void update(NotificationTemplateDto notificationTemplateDto) {
             .descending());
         Page<NotificationTemplate> notificationTemplates = notificationTemplateRepository.findAll(pageRequest);
         List<NotificationTemplateDto> templateDtoList = notificationTemplates.stream()
-            .map(x -> modelMapper.map(x, NotificationTemplateDto.class)).collect(Collectors.toList());
+            .map(x -> modelMapper.map(x, NotificationTemplateDto.class)
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java]: Try to find better name for variable, better to use name like templateDto

**Review by VadymMakitra — CHANGES_REQUESTED** [review]
Please resolve comments

File: service/src/main/java/greencity/service/NotificationServiceImpl.java (hunk 2 above)
Comment (DimaSannytski) [reply service/src/main/java/greencity/service/NotificationServiceImpl.java:355]: No

File: service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java (hunk 3 above)
Comment (DimaSannytski) [reply service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java]: I think if i have list of dtos i like to give them name dtolist more than just dto

File: service/src/main/java/greencity/service/NotificationServiceImpl.java (hunk 2 above)
Comment (DimaSannytski) [reply service/src/main/java/greencity/service/NotificationServiceImpl.java:355]: We have not dto for mapping

File: service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java (hunk 3 above)
Comment (VadymMakitra) [reply service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java]: @DimaSannytski I mean rename x

Comment (DimaSannytski) [reply service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java]: .map(x( is dto)->mod.map(x( is entity)) in my opinion it's not the best idea rename x to templatedto

Comment (thesvider) [reply service/src/main/java/greencity/service/notification/ManagementNotificationServiceImpl.java]: agree. please avoid names like "X"

**Review by thesvider — APPROVED** [review]

**Review by VadymMakitra — APPROVED** [review]

**Pull request merged**